be2-scala/src/test/scala/ch/epfl/pop/pubsub/graph/validator/Lao/CreateLaoDecoderSuite.scala

```diff
@@ -0,0 +1,110 @@
+package ch.epfl.pop.pubsub.graph.validator.lao
+
+import org.scalatest.FlatSpec
+import org.scalatest.Matchers
+import ch.epfl.pop.model.network.method.message.Message
+import ch.epfl.pop.pubsub.graph.GraphMessage
+import ch.epfl.pop.model.network.JsonRpcRequest
+import ch.epfl.pop.model.network.MethodType
+import ch.epfl.pop.model.network.method.ParamsWithMessage
+import ch.epfl.pop.model.objects.Channel
+import util.examples.CreateLaoExamples
+import ch.epfl.pop.pubsub.graph.MessageDecoder
+import ch.epfl.pop.model.network.requests.lao.JsonRpcRequestCreateLao
+import org.scalatest.GivenWhenThen
+import ch.epfl.pop.model.network.requests.lao.JsonRpcRequestStateLao
+import ch.epfl.pop.model.network.method.message.data.lao.CreateLao
+import ch.epfl.pop.pubsub.graph.ErrorCodes
+import ch.epfl.pop.pubsub.graph.PipelineError
+import org.scalatest.Inside
+
+class CreateLaoDecoderSuite extends FlatSpec with Matchers with Inside with GivenWhenThen {
+
+  def withCreateLaoFixiture(msg: Message)(testCode: (GraphMessage, Message) => Any){
+    val jsonReq = CreateLaoExamples.getJsonRequestFromMessage(msg)
+    testCode(Left(jsonReq),msg)
+  }
+
+  def testGoodFormat =
+      (gm: GraphMessage, createLaoMessage: Message) => {
+        Given("a correct graph message of JsonRpcRequest")
+        //gm
+        And("a createLao message")
+        //createLaoMessage
```

Comment:
What do those comments do? I'm not familiar with this syntax

Reply:
Just to give provide extra information along with the Given & And. 
Do you find them noisy I can delete them if so